### ISSUE TITLE
reportUnusedVariable considers variable as used after reassignment

### ISSUE THREAD

**Issue by solutionseekeras** [microsoft/pyright#8806]
**Describe the bug**

Pyright considers a variable to be used if it has been used at least once, even if it is later reassigned to without use.

**Code or Screenshots**

```python

def f() -> None:
    a = 1

    # I would expect the left-hand side `a` to be flagged as unused here (PyCharm does so)
    a = 2 * a

# pyright: reportUnusedVariable=error
```

`pyright <path-to-file>` results in `0 errors, 0 warnings, 0 informations`.

**VS Code extension or command-line**
pyright 1.1.377 on the command-line.

**Comment by erictraut** [comment]
This isn't how `reportUnusedVariable` works in pyright. It is reported if the variable is not consumed anywhere in the function.

Pyright is working as designed here, so closing the issue.

**Comment by solutionseekeras** [comment]
Sounds fair, thanks for the quick response.

Are there any plans to add detection of this? If not, what's the reasoning?
